sympy/solvers/ode.py

```diff
@@ -4089,6 +4089,25 @@ def ode_nth_order_reducible(eq, func, order, match):
 
     return fsol
 
+# This needs to produce an invertible function but the inverse depends
+# which variable we are integrating with respect to. Since the class can
+# be stored in cached results we need to ensure that we always get the
+# same class back for each particular integration variable.
+def _nth_algebraic_diffx(var, _stored={}):
+    cls = _stored.get(var, None)
```

Comment:
It seems that `_stored` is not saved anywhere. Won't it be lost when this function returns?

Reply:
`_stored` is the default argument for the function and persists across calls. This is an often misunderstood feature of Python and is the reason that mutable defaults are *normally* best avoided. Here's a demonstration:
```julia
In [2]: cat q.py                                                                                                                  
def square(x, _store={}):
    print(_store)
    if x not in _store:
        _store[x] = x**2
    return _store[x]

In [3]: square(3)                                                                                                                 
{}
Out[3]: 9

In [4]: square(4)                                                                                                                 
{3: 9}
Out[4]: 16

In [5]: square(5)                                                                                                                 
{3: 9, 4: 16}
Out[5]: 25

In [6]: square.__defaults__                                                                                                       
Out[6]: ({3: 9, 4: 16, 5: 25},)
```

Reply:
I didn't know this. Is it described somewhere in the documentation? (So it is not implementation dependent.)

Reply:
It is explained here:
https://docs.python.org/3/reference/compound_stmts.html#function-definitions

"Default parameter values are evaluated from left to right when the function definition is executed...."

Reply:
Would it be possible to make `_stored` global?

Reply:
Yes. It already is global in the sense that there is only one and it is attached to the function which is itself global but I can just make it into a regular global variable.

Reply:
I would prefer making it an explicit global variable. That would make the code more understandable.